numpy/ma/core.py

```diff
@@ -3316,6 +3317,11 @@ def _scalar_heuristic(arr, elem):
                                 f"all to {dout._fill_value[0]!s}.",
                                 stacklevel=2)
                         dout._fill_value = dout._fill_value.flat[0]
```

Comment:
This line looks like it has a bug too, sadly. I'll see if I can produce a test-case at some point.